### ISSUE TITLE
Multi-Queue Implementation

### ISSUE THREAD

**Issue by cwfitzgerald** [gfx-rs/wgpu#1066]
**Is your feature request related to a problem? Please describe.**

We should make a testing implementation of https://github.com/gpuweb/gpuweb/pull/1169.

**Describe the solution you'd like**

Multi-queue.

**Describe alternatives you've considered**

Single-queue.

**Additional context**

There's a plethora of issues you can solve with multi-queue, though they are an advanced feature.

**Comment by DrRuhe** [comment]
Any updates on this? I'd be interested in helping push this along.

**Comment by pingpong74** [comment]
Hi, Any updates on this? I would like to contribute to this.

**Comment by Swoorup** [comment]
Curious what backend supports this already? Metal, OpenGL, Directx? (doubt its webgl, or webgpu)

**Comment by pingpong74** [comment]
I think only Vulkan and Directx 12 support this.

**Comment by cwfitzgerald** [comment]
Yeah, both DirectX and Vulkan support it. Metal is more complicated because it actually has three hardware queues (vertex, fragment, and compute). But you can use as many software queues as you want to run work on those three hardware queues. So we will still support Multi-Queue on Metal.
